test/test_ops.py

```diff
@@ -1102,296 +1101,382 @@ def test_bbox_convert_jit(self):
         )
 
         scripted_fn = torch.jit.script(ops.box_convert)
-        TOLERANCE = 1e-3
+        atol = 1e-3
 
         box_xywh = ops.box_convert(box_tensor, in_fmt="xyxy", out_fmt="xywh")
         scripted_xywh = scripted_fn(box_tensor, "xyxy", "xywh")
-        torch.testing.assert_close(scripted_xywh, box_xywh, rtol=0.0, atol=TOLERANCE)
+        torch.testing.assert_close(scripted_xywh, box_xywh, rtol=0.0, atol=atol)
 
         box_cxcywh = ops.box_convert(box_tensor, in_fmt="xyxy", out_fmt="cxcywh")
         scripted_cxcywh = scripted_fn(box_tensor, "xyxy", "cxcywh")
-        torch.testing.assert_close(scripted_cxcywh, box_cxcywh, rtol=0.0, atol=TOLERANCE)
+        torch.testing.assert_close(scripted_cxcywh, box_cxcywh, rtol=0.0, atol=atol)
 
 
-class BoxTestBase(ABC):
-    @abstractmethod
-    def _target_fn(self) -> Tuple[bool, Callable]:
-        pass
+class TestBoxArea:
+    def area_check(self, box, expected, atol=1e-4):
+        out = ops.box_area(box)
+        torch.testing.assert_close(out, expected, rtol=0.0, check_dtype=False, atol=atol)
+
+    @pytest.mark.parametrize("dtype", [torch.int8, torch.int16, torch.int32, torch.int64])
+    def test_int_boxes(self, dtype):
+        box_tensor = torch.tensor([[0, 0, 100, 100], [0, 0, 0, 0]], dtype=dtype)
+        expected = torch.tensor([10000, 0], dtype=torch.int32)
+        self.area_check(box_tensor, expected)
+
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+    def test_float_boxes(self, dtype):
+        box_tensor = torch.tensor(
+            [
+                [285.3538, 185.5758, 1193.5110, 851.4551],
+                [285.1472, 188.7374, 1192.4984, 851.0669],
+                [279.2440, 197.9812, 1189.4746, 849.2019],
+            ],
+            dtype=dtype,
+        )
+        expected = torch.tensor([604723.0806, 600965.4666, 592761.0085], dtype=dtype)
+        self.area_check(box_tensor, expected, atol=0.05)
 
-    def _perform_box_operation(self, box: Tensor, run_as_script: bool = False) -> Tensor:
-        is_binary_fn = self._target_fn()[0]
-        target_fn = self._target_fn()[1]
-        box_operation = torch.jit.script(target_fn) if run_as_script else target_fn
-        return box_operation(box, box) if is_binary_fn else box_operation(box)
+    def test_float16_box(self):
+        box_tensor = torch.tensor(
+            [[285.25, 185.625, 1194.0, 851.5], [285.25, 188.75, 1192.0, 851.0], [279.25, 198.0, 1189.0, 849.0]],
+            dtype=torch.float16,
+        )
+        expected = torch.tensor([605113.875, 600495.1875, 592247.25], dtype=torch.float32)
+        self.area_check(box_tensor, expected)
```

Comment:
As @NicolasHug highlighted above, we are trying to check float16 values that are caped at 65k against expected values at around 600k. This shouldn't work. We should revise the testing values to avoid overflows.

Reply:
Yep. I understood now. Let me  revise the tests,